src/java/fr/paris/lutece/plugins/forms/web/FormXPage.java

```diff
@@ -1129,20 +1129,21 @@ public XPage doSynchronousUploadDocument( HttpServletRequest request ) throws Si
      */
     private void saveFormResponse( Form form, HttpServletRequest request ) throws SiteMessageException
     {
-        /*
-         * The deletion of the synchronized block is due to the accumulation of http threads on the server when scaling up. The current implementation does not
-         * guarantee the following two checks in case of competitor access: -checkNumberMaxResponseForm (form, request); -checkIfUserResponseForm (form,
-         * request); This implementation to review and modify.
-         */
-        checkNumberMaxResponseForm( form, request );
-        checkIfUserResponseForm( form, request );
         FormResponse formResponse = _formResponseManager.getFormResponse( );
         if ( form.isAuthentificationNeeded( ) )
         {
+            checkIfUserResponseForm( form, request );
```

Comment:
Doesn't this need some form of synchronisation ?
One way of ensuring this would be to add a unique column in the DB. For instance, add a table with columns `(userid, formid)`, with a primary key spanning both columns, and filled for forms where a single response is mandated.